File: frontend/src/services/patientService.js
import api from "./api";

export const getPatients = () => {
    return api.get("/patients");
};

export const addPatient = (patient) => {
    return api.post("/patients", patient);
};

export const updatePatient = (id, patient) => {
    return api.put(`/patients/${id}`, patient);
};

export const deletePatient = (id) => {
    return api.delete(`/patients/${id}`);
};

File: backend/src/main/java/hospital_management/backend/service/PatientService.java
package hospital_management.backend.service;

import hospital_management.backend.entity.Patient;

import java.util.List;

public interface PatientService {

    Patient addPatient(Patient patient);

    List<Patient> getAllPatients();

    Patient getPatientById(Long id);

    Patient updatePatient(Long id, Patient patient);

    void deletePatient(Long id);

}